fix: buy interface changed with cairo 2 update

Th0rgal opened the pull request:
This pull request fixes an issue in the Cairo 2 updated frontend. We didn't follow the correct buy interface, thus passing the targetAddress (which was removed in Cairo 2) as the sponsor param.

With the cairo 2 update we removed the targetAddress field in the buy. Instead by default a domain will resolve to his starknet id owner. If we wanted to make sure the targetAddress is something specific (not his owner), we could still do it with a multicall on the starknetid, that would be much cleaner.

fricoben commented on components/domains/registerV2.tsx:
```diff
@@ -188,8 +188,10 @@ const RegisterV2: FunctionComponent<RegisterV2Props> = ({ domain, groups }) => {
+    console.log("chosing sponsor!");
     if (referralData) {
       const data = JSON.parse(referralData);
+      console.log("data:", data.sponsor);
```
delete logs

fricoben commented on components/domains/registerV2.tsx:
```diff
@@ -214,12 +216,13 @@ const RegisterV2: FunctionComponent<RegisterV2Props> = ({ domain, groups }) => {
       ),
     ];
+    console.log("sponsor:", sponsor);
+    console.log("calls:", calls);
```
uh

fricoben reviewed (changes requested):
- [x] needs to delete logs
- [ ] Needs to delete target address here then 

<img width="663" alt="Capture d’écran 2024-03-11 à 14 12 30" src="https://github.com/starknet-id/app.starknet.id/assets/78437165/91b4b5e6-cecf-4182-a862-2fbdc3d0ea0a">

Th0rgal committed:
fix: buy interface changed with cairo 2 update

Th0rgal committed:
fix: remove set target address

Th0rgal commented:
Update: as you suggested I removed the option to change the target address. This means that in the implementation I removed the test checking if the target address is the owner address (by considering it as always true).

irisdv reviewed (approved):
lgtm

fricoben reviewed (approved):
lgtm
